00_Base/src/datasources/ITariffsDatasource.ts

```diff
@@ -0,0 +1,21 @@
+import { TariffDTO } from '../model/DTO/tariffs/TariffDTO';
+import { OcpiTariff, TariffKey } from '../model/OcpiTariff';
+import { GetTariffsParams } from '../model/DTO/tariffs/GetTariffsParams';
+import { PutTariffRequest } from '../model/DTO/tariffs/PutTariffRequest';
+
+export interface ITariffsDatasource {
+  getTariffByKey(
+    key: TariffKey,
+    isCoreTariffKey?: boolean,
```

Comment:
I had similar thing with Tokens: https://github.com/citrineos/citrineos-ocpi/pull/114.

The `BaseTransactionMapper.ts` needed to fetch Token by the OCPP model `IdToken`.

So it needed to call a method declared as:
```
  async getTokenDtoByIdToken(
    idToken: string,
    type: IdTokenEnumType,
  ): Promise<TokenDTO | undefined>
```

Which is problematic since it contains OCPP references.

Instead of putting this declaration onto `ITokensDatasource`, I only put it onto the `TokensDatasource`.

I think this makes sense as the implementation itself can be OCPP specific, but the interface should not be. This solution seems to achieve exactly that.

Reply:
Also, in such case, the TokensDatasource is explicitly wired in the `BaseTransactionMapper.ts` instead of declaring the interface.

This makes sense as `BaseTransactionMapper.ts` itself is OCPP specific so it makes sense to wire in the exact datasource implementation that matches (i.e. an OCPP 201 mapper should wire in an OCPP 201 datasource).

Reply:
Got it makes sense, we want to keep it clean as possible 🤔 Good suggestion! Let me check it out!

Reply:
@langiole Just pushed new implementation, let me know what you think!

I was thinking it'd be worth keeping the interface abstraction still since if we move to different versions, I'm guessing we'd still need some kind of mapping between OCPI Tariff and OCPP Tariff so that we can map them together to avoid data duplication 🤔

Reply:
I see what you're saying, but we are not always guaranteed 100% overlap between OCPP versions.

For example, in your `ICoreTariffsDatasource` you have defined:
```
getTariffByCoreKey(key: TariffKey): Promise<TariffDTO | undefined>;
```

Like you said, it is likely that this exact method will be useful for both OCPP versions.

But take a look at `TokensDatasource` in the tokens PR and you will see this method:
```
async getTokenDtoByIdToken(idToken: string, type: IdTokenEnumType,): Promise<TokenDTO | undefined>
```

You may think such method should be defined in a `ICoreTokensDatasource`. The issue is that the `idToken` arg is only relevant for OCPP 201 and not 1.6. So if you have 1.6 implementation of `ICoreTokensDatasource`, you will not be able to fill the implementation for this method out since `IdToken` is not a thing in 1.6.

Reply:
I think for now it is best to keep it simple where we explicitly call the datasource implementation as in the tokens PR.

I wouldn't consider this bad practice since the caller is already OCPP version specific. I.e. `BaseTransactionMapper` is OCPP 201 specific so it makes sense that it explicitly wires in a 201 instance of `ITokensDatasource`.